cli/medperf/entities/interface.py

```diff
@@ -1,77 +1,215 @@
 from typing import List, Dict, Union
-from abc import ABC, abstractmethod
+from abc import ABC
+import logging
+import os
+import yaml
+from medperf.exceptions import MedperfException, InvalidArgumentError
+from medperf.entities.schemas import MedperfBaseSchema
 
 
-class Entity(ABC):
-    @abstractmethod
-    def all(
-        cls, local_only: bool = False, comms_func: callable = None
-    ) -> List["Entity"]:
+class Entity(MedperfBaseSchema, ABC):
+    @staticmethod
+    def get_type():
+        raise NotImplementedError()
+
+    @staticmethod
+    def get_storage_path():
+        raise NotImplementedError()
+
+    @staticmethod
+    def get_comms_retriever():
+        raise NotImplementedError()
+
+    @staticmethod
+    def get_metadata_filename():
+        raise NotImplementedError()
+
+    @staticmethod
+    def get_comms_uploader():
+        raise NotImplementedError()
+
+    @property
+    def identifier(self):
+        return self.id or self.generated_uid
```

Comment:
There is no `id` or `generated_uid` in neither `Entity` or `MedperfBaseSchema`. But there is `id` in `MedperfSchema`.
It seems that `id` should present everywhere so I'd propose to move it from `MedperfSchema` to `Entity` directly;
while `generated_uid` does not exist say for dataset, so it would be better if any entity that uses `generated_uid` overwrite this property on its own

Reply:
Hmm. I noticed in old code `Dataset` had both `id` and `generated_uid` while in new code it seems to not have id?..
Also TestResult seems not to have `id` but `generated_uid` only?.. Anyway, it's worth to declare explicitly these fields

Reply:
we were using `generated_uid` in two different contexts, which makes it confusing:
The dataset object has in its Server schema a `generated_uid`, which actually should be really renamed to `output_hash` but such renaming may cause us pain now.
All entities have some sort of "local ID", which we were refering to by `generated_uid`.
Now, I pushed a change that removes such an ambiguity by renaming the general `generated_uid` to `local_id`. Dataset's `local_id` happens to be the `generated_uid`, but I hope that this is not confusing anymore (may be confusing to people who already read the code before this change, but I think it's clearer for people reading the code for the first time).
and also I defined a `local_id` in the superclass.